### PR TITLE
when dimvalue`s len > short.max_value then [Extract Fact Table Distinct Columns] err

### PR DESCRIPTION
**Pull request by javalife0312**
https://issues.apache.org/jira/browse/KYLIN-3775

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: core-dictionary/src/main/java/org/apache/kylin/dict/TrieDictionary.java [hunk 1]
```diff
@@ -89,6 +89,35 @@ public TrieDictionary(byte[] trieBytes) {
...
+        try {
+            this.maxValueLength = headIn.readInt();
+            if (maxValueLength < 0) {
+                throw new IllegalStateException("maxValueLength is negative (" + maxValueLength
+                        + "). Dict value is too long, whose length is larger than " + Integer.MAX_VALUE);
+            }
+            String converterName = headIn.readUTF();
+
+            return converterName;
+        } catch (Exception e) {
```
Comment (nichunen) [inline core-dictionary/src/main/java/org/apache/kylin/dict/TrieDictionary.java]: When will this happen?

**Comment by nichunen** [comment]
@javalife0312 Hi, can you open a jira, and describe the issue troubled  you in detail?

**Comment by nichunen** [comment]
After discussed with @javalife0312  
We think this is not a common case, and there is workaround way to deal with long dimensions.
Close this pr

**Pull request closed without merging**